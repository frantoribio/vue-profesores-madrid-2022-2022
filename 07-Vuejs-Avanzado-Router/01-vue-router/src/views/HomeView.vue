<template>
  <main>
    <p>Home</p>
    <p>Vamos a navegar a mano</p>
    <button @click="goToAcercaDe">Ir a Acerca De</button>
    <button @click="goToDetails(2)">Ir a detalles 2</button>
  </main>
</template>
<script setup>
  import { useRouter } from 'vue-router'

  const router = useRouter()

  const goToAcercaDe = () => router.push({ name: 'about' })
  const goToDetails = (index) => router.push({ name: 'details', params: { id: index } })
</script>
