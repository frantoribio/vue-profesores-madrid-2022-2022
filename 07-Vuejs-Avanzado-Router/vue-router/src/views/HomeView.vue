<template>
  <main>
    <p>Home</p>
  </main>
</template>
<script setup>
</script>
