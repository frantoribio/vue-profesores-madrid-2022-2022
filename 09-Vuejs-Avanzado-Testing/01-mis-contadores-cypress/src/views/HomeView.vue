<template>
  <main>
    <h1>Home</h1>
    <p>Home page</p>
    <hr />
    <Contador :valorInicial="10" />
  </main>
</template>

<script setup>
  import Contador from '@/components/Contador.vue'
</script>
