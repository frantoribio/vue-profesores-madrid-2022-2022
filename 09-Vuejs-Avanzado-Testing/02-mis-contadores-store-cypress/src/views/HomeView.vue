<template>
  <main>
    <h1>Home</h1>
    <p>Home page</p>
    <hr />
    <Contador :valorInicial="10" />
    <input
      type="text"
      v-model="nombre"
      data-cy="input-nombre"
    />
    <p data-cy="texto-nombre">{{ nombre }}</p>
  </main>
</template>

<script setup>
  import Contador from '@/components/Contador.vue'
  import { ref } from 'vue'

  const nombre = ref('')
</script>
