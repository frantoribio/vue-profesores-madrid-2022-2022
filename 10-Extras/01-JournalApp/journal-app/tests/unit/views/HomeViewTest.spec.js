import { beforeEach, describe, expect, test, vi } from 'vitest'

import HomeView from '@/views/HomeView.vue'
import { shallowMount } from '@vue/test-utils'

import { createRouterMock, injectRouterMock } from 'vue-router-mock'

// Vamos a usar https://github.com/posva/vue-router-mock#testing-libraries

describe('View -> HomeView', () => {
  // Porque e sVitest debemos configurarlo así
  // const router = createRouterMock({
  //   spy: {
  //     create: (fn) => vi.fn(fn),
  //     reset: (spy) => spy.mockReset(),
  //   },
  // })
  // Si usamos globals no es necesario todo esto
  const router = createRouterMock()

  let wrapper

  beforeEach(() => {
    // inject it globally to ensure `useRoute()`, `$route`, etc work
    // properly and give you access to test specific functions
    injectRouterMock(router)
    wrapper = shallowMount(HomeView)
  })

  test('Debe hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot() // snapshot es una imagen de la vista y podemos usar .element tambien o html()
  })

  test('Debe hacer click el botón de Daybook', async () => {
    // Mockeramos el método
    wrapper.vm.goToDaybook = vi.fn()

    const dayButton = wrapper.find('[data-testid="DayButton"]')
    expect(dayButton.exists()).toBe(true)

    expect(dayButton.text()).toBe('DayBook')
    await dayButton.trigger('click')

    expect(wrapper.vm.goToDaybook).toHaveBeenCalledTimes(1)
  })

  test('Debe hacer click el botón de About', async () => {
    // Mockeramos el método
    wrapper.vm.goToAbout = vi.fn()

    const dayButton = wrapper.find('[data-testid="AboutButton"]')
    expect(dayButton.exists()).toBe(true)

    expect(dayButton.text()).toBe('About')
    await dayButton.trigger('click')

    expect(wrapper.vm.goToAbout).toHaveBeenCalledTimes(1)
  })

  test('debe hacer click en el botón debe redireccionar a daybook-no-entry', async () => {
    // Trabajamos con el boton
    const dayButton = wrapper.find('[data-testid="DayButton"]')
    expect(dayButton.exists()).toBe(true)

    expect(dayButton.text()).toBe('DayBook')
    await dayButton.trigger('click')

    expect(router.push).toHaveBeenCalledWith({ name: 'daybook-no-entry' })
  })

  test('Debe hacer click en el botón debe redireccionar a about', async () => {
    // Trabajamos con el boton
    const dayButton = wrapper.find('[data-testid="AboutButton"]')
    expect(dayButton.exists()).toBe(true)

    expect(dayButton.text()).toBe('About')
    await dayButton.trigger('click')

    expect(router.push).toHaveBeenCalledWith({ name: 'about' })
  })
})
