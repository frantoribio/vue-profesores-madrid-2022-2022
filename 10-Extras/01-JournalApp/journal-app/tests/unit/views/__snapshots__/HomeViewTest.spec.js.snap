// Vitest Snapshot v1

exports[`View -> HomeView > Debe hacer match con el snapshot 1`] = `
"<div data-theme=\\"joseluisgs\\" class=\\"container mx-auto rounded-lg px-3 py-4 text-center shadow-xl\\">
  <h1 class=\\"cursor-pointer text-xl font-semibold text-gray-500 hover:text-indigo-500\\"> This is an home page </h1><button data-testid=\\"DayButton\\" class=\\"btn btn-primary\\"> DayBook </button><button data-testid=\\"AboutButton\\" class=\\"btn btn-secondary\\"> About </button><button class=\\"btn btn-outline btn-success\\"> Abrir Modal </button>
  <article class=\\"prose mx-auto text-center\\">
    <h1>Garlic bread with cheese: What the science tells us</h1>
    <p> Lorem, ipsum dolor sit amet consectetur adipisicing elit. Repudiandae minus quam harum ab dolorum laudantium perspiciatis dolores corporis commodi, a quidem. Possimus ipsum magnam repellendus ipsam laboriosam architecto mollitia consectetur?. </p>
    <p> Lorem ipsum dolor sit, amet consectetur adipisicing elit. Accusantium eveniet magni deleniti eius. Nisi ipsa accusantium voluptatem quis unde sunt, ratione itaque eius atque quia odio ab aperiam explicabo! Laboriosam!. </p><!-- ... -->
  </article>
</div>
<!-- Mi modal -->
<!--v-if-->"
`;
